ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/desc/DescTableOperation.java

```diff
@@ -148,7 +148,8 @@ private void getColumnsNoColumnPath(Table table, Partition partition, List<Field
 
     // Fetch partition statistics only for describe extended or formatted.
     if (desc.isExtended() || desc.isFormatted()) {
-      if (table.isPartitioned() && partition == null) {
+      boolean shouldGetPartStats = MetastoreConf.getBoolVar(context.getConf(), MetastoreConf.ConfVars.DESCTABLE_ENABLE_PARTITION_STATS);
```

Comment:
We need to ensure the stats field itself is removed in the output if this config is enabled. Also, keep it uniform for both table and partition stats. Returning incorrect stats misleads user.

Reply:
Added the unit test. Please comment on what this particular line needs if any in the updated PR.